internal/services/controller/delta.go

```diff
@@ -94,12 +94,14 @@ func (d *delta) toCASTAIRequest() *castai.Delta {
 	}
 }
 
-func encode(obj interface{}) ([]byte, error) {
+func encode(obj interface{}) (*json.RawMessage, error) {
 	b, err := json.Marshal(obj)
 	if err != nil {
 		return nil, fmt.Errorf("marshaling %T to json: %v", obj, err)
 	}
-	return b, nil
+	// it should allow sending raw json over network without being encoded to base64
+	o := json.RawMessage(fmt.Sprintf(`%v`, string(b)))
```

Comment:
`json.RawMessage` is a type alias: `type RawMessage []byte`, so  you don't need to convert it to string and to byte again. Just return the `[]byte` you get from marshaling the object.

Reply:
yeah, you are right) removed unneeded conversation to `string` and back, but need to return `json.RawMessage` instead of []byte to avoid base64 encoding of `[]byte` during the marshalling operation